kv/union_store.go

```diff
@@ -141,45 +154,36 @@ func (us *unionStore) Get(k Key) ([]byte, error) {
 }
 
 // markLazyConditionPair marks a kv pair for later check.
-func (us *unionStore) markLazyConditionPair(k Key, v []byte) error {
-	if len(v) == 0 {
-		return errors.Trace(us.lazyConditionPairs.Delete(k))
+func (us *unionStore) markLazyConditionPair(k Key, v []byte, e error) error {
+	us.lazyConditionPairs[string(k)] = &conditionPair{
+		key:   k,
```

Comment:
The Key should be cloned before store in map, the caller may change the content after calling `Get()`.
Or maybe we can use the map key directly in later check, then no need to store it here.